src/main/java/com/jcabi/http/wire/AbstractHeaderBasedCachingWire.java

```diff
@@ -173,13 +171,15 @@ private Response validateCacheWithServer(
     }
 
     /**
-     * Add response to cache.
+     * Add, update or evict response in cache.
      * @param req The request to be used as key
-     * @param rsp The response to add
+     * @param rsp The response to add/update
      */
     private void updateCache(final Request req, final Response rsp) {
         if (rsp.headers().containsKey(this.scvh)) {
             this.cache.put(req, rsp);
+        } else {
```

Comment:
@mkordas I did wrongly assume that there's no way `updateCache` is being called in event of non-OK,but turns out I was wrong. In my defense however the task does _not_ specify what I should do WRT eviction in any other case. i.e. the existing tests pass and the new specified scenario condition passes, other scenarios are untested and unspecified edge cases. I can implement this alongside another test, if you're sure that's the desired behavior?